### DISCUSSION TITLE
Logging Line number, method name, and file path

### DISCUSSION THREAD

**Discussion by Tthongsy** [datalust/seq-tickets#1335] [Help] [answered]
Hello,

I was curious if there was an enricher that allowed me to log the line number, method name, and file path.
I used reflection originally, but I was wondering if there was another way that wasn't with poor performance?

Thanks Tommy T

**Comment by nblumhardt** [comment, marked as answer]
Hi Tommy,

You can avoid reflection if you're able to change the call sites in a way that supports `CallerMemberInfo`:

https://stackoverflow.com/questions/48268854/how-to-get-formatted-output-from-logevent/48272467#48272467

Otherwise, it's reflection and poor performance, unfortunately.

**Reply by Tthongsy** [reply]
Thanks!

**Comment by MattMofDoom** [comment]
@Tthongsy FWIW I've implemented 'automatic' logging of line number, method name, and file path as part of [Lurgle.Logging](https://github.com/MattMofDoom/Lurgle.Logging), which is a library implementation of Serilog that bundles a number of sinks and enrichers (and implements a couple more for added functionality). This is done by having a call to (for example) Log.Information().Add("A log entry"), or Log.Level(LurgLevel.Information).Add("A log entry"), where the callermemberinfo is integrated into the first (static) method. 

There's a bunch of features like masking, correlation ids, etc, but the main reason I mention it is that even if you don't use Lurgle, there's some ideas in there that might give you some direction.
